File: 0219-contains-duplicate-ii/0219-contains-duplicate-ii.cs
public class Solution {
    public bool ContainsNearbyDuplicate(int[] nums, int k) {
        int start = 0;
        int end = 0;
        
        HashSet<int> mySet = new HashSet<int>();
        
        while(end < nums.Length){
            
            if(end - start > k){
                mySet.Remove(nums[start]);
                start++;
            }
            
            if(!mySet.Add(nums[end])){
                return true;
            }
            
            mySet.Add(nums[end]);
            end++;
        }
        
        return false;
    }
}

File: 0219-contains-duplicate-ii/0219-contains-duplicate-ii.java
class Solution {
    public boolean containsNearbyDuplicate(int[] nums, int k) {
        int start = 0;
        int end = 0;
        
        Set<Integer> mySet = new HashSet<>();
        
        while(end < nums.length){
            
            if(end - start > k){
                mySet.remove(nums[start]);
                start++;
            }
            
            if(!mySet.add(nums[end])){
                return true;
            }
            
            mySet.add(nums[end]);
            end++;
        }
        return false;
    }
}